corehq/apps/events/views.py

```diff
@@ -377,22 +383,84 @@ class AttendeesConfigView(JSONResponseMixin, BaseUserSettingsView, BaseEventView
-            "mobile_worker_attendee_enabled": mobile_worker_attendees_enabled(self.domain)
+            'mobile_worker_attendee_enabled': mobile_worker_attendees_enabled(self.domain)
         })
 
+
+class ConvertMobileWorkerAttendeesView(BaseUserSettingsView, BaseEventView):
+    urlname = "convert_mobile_workers"
+
     @allow_remote_invocation
-    def post(self, request, *args, **kwargs):
-        json_data = json.loads(request.body)
-        attendees_enabled = json_data['mobile_worker_attendee_enabled']
+    def get(self, request, *args, **kwargs):
+        attendees_enabled = mobile_worker_attendees_enabled(self.domain)
         toggle_mobile_worker_attendees(self.domain, attendees_enabled)
+
+        task_ref = expose_cached_download(
+            payload=None, expiry=60 * 60, file_extension=None
+        )
+
         if attendees_enabled:
-            sync_mobile_worker_attendees.delay(self.domain, user_id=self.couch_user.user_id)
+            task = sync_mobile_worker_attendees.delay(self.domain, user_id=self.couch_user.user_id)
         else:
-            close_mobile_worker_attendee_cases.delay(self.domain)
+            task = close_mobile_worker_attendee_cases.delay(self.domain)
 
-        return self.json_response({
-            "mobile_worker_attendee_enabled": attendees_enabled
+        task_ref.set_task(task)
+        return HttpResponseRedirect(
+            reverse(
+                MobileWorkerAttendeeSatusView.urlname,
+                args=[self.domain, task_ref.download_id]
+            )
+        )
+
+
+class MobileWorkerAttendeeSatusView(BaseEventView):
+    urlname = "convert_mobile_worker_status"
+
+    @property
+    def page_title(self):
+        if mobile_worker_attendees_enabled(self.domain):
+            return _("Disabling mobile worker attendees status")
+        else:
+            return _("Enabling mobile worker attendees status")
+
+    def get(self, request, *args, **kwargs):
+        context = super(MobileWorkerAttendeeSatusView, self).main_context
+        context.update({
+            'domain': self.domain,
+            'download_id': kwargs['download_id'],
+            'poll_url': reverse('poll_mobile_worker_attendee_progress', args=[self.domain, kwargs['download_id']]),
+            'title': _(self.page_title),
+            'progress_text': _("Converting Users. This may take some time..."),
+            'error_text': _("User conversion failed for some reason and we have noted this failure."),
         })
+        return render(request, 'hqwebapp/soil_status_full.html', context)
+
+    def page_url(self):
+        return reverse(self.urlname, args=self.args, kwargs=self.kwargs)
+
+
+@require_GET
+@login_and_domain_required
+def poll_mobile_worker_attendee_progress(request, domain, download_id):
+    try:
+        context = get_download_context(download_id)
+    except TaskFailedError:
+        return HttpResponseServerError()
+
+    if mobile_worker_attendees_enabled(domain):
```

Comment:
Instead of using a conditional block for enabling and disabling, you could create a variable:
```
status_text = "Disabling"
if mobile_worker_attendees_enabled(self.domain):
    status_text = "Enabling"
```
You can then simply have the messages as `context['success_message'] = _(f"{status_text} mobile workers complete!")` for example.

Reply:
> `_(f"{status_text} mobile workers complete!")` 

Careful: `_` (i.e. `gettext_lazy()` here) [cannot take f-strings](https://docs.djangoproject.com/en/3.2/topics/i18n/translation/).

And if you used `format()` instead, like ...
```python
    status = _('Enabling') if enabled else _('Disabling')
    msg = _("{status} mobile workers complete!").format(status=status)
```
... you still have the problem of assuming a sentence structure that might not be true for all languages (e.g. if the French translation is "Les travailleurs mobiles sont activés" (I wouldn't know :shrug: this is just a made-up example) then a translator is not going to be able to figure out how to make the sentences make sense in their language.

Best to use `gettext()` on whole sentences (or paragraphs) at a time.